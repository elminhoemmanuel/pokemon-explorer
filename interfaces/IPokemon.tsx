export interface IPokemonsEach {
    name: string,
    img: string
}

export interface IPokemonsResult {
    name: string,
    url: string
}
export interface IResult {
    count: number,
    next: string | null,
    previous: string | null,
    results: IPokemonsResult[]
}

export interface IFetch{ 
    loading:boolean,
    error:string
}

export interface IBasic {
    name:string,
    url:string
}

export interface IStats{
    base_stat:number,
    effort:0,
    stat:IBasic
}
export interface IType{
    slot:number,
    type:IBasic
}
export interface IVersion{
    level_learned_at:number,
    move_learn_method:IBasic,
    version_group:IBasic
}

export interface IMove{
    move:IBasic,
    version_group_details:IVersion[]
}

export interface IPokemonDetails{ 
    species:IBasic,
    stats:IStats[],
    types:IType[],
    weight:number,
    moves:IMove[],
    [extra:string]:any

}

export interface IPagination{
    next:string | null,
    previous:string | null,
    nextPage:()=>void,
    prevPage:()=>void,
}
