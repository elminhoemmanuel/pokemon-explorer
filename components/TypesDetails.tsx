import React from 'react'
import { IType } from 'interfaces/IPokemon'

type TypesListProps = {
    header: string,
    iterator: IType[]
}

const TypesList = ({ header, iterator }: TypesListProps) => {
    return (
        <div>
            <p><span className="font-bold underline text-lg">{header}</span> :</p>
            <div className="pl-4 mb-6">
                {
                    iterator?.map(item => (
                        <p key={item.type.name} >{item.type.name}</p>
                    ))
                }
            </div>
        </div>
    )
}

export default TypesList