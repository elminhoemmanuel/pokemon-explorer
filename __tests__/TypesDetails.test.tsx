import { render, screen } from '@testing-library/react'
import TypesDetails from '../components/TypesDetails'
import { within } from '@testing-library/dom'
import { IType } from "../interfaces/IPokemon"

const testTypes: IType[] = [
    {
        "slot": 1,
        "type": {
            "name": "grass",
            "url": "https://pokeapi.co/api/v2/type/12/"
        }
    },
    {
        "slot": 2,
        "type": {
            "name": "poison",
            "url": "https://pokeapi.co/api/v2/type/4/"
        }
    }

]



test("On render, check that types is displayed correctly", () => {
    const { container } = render(
        <TypesDetails header="Types" iterator={testTypes} />
    );
    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toMatchSnapshot()
})