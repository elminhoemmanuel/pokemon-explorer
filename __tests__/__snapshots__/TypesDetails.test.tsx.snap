// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On render, check that moves is displayed correctly 1`] = `
<div>
  <p>
    <span
      class="font-bold underline text-lg"
    >
      Moves
    </span>
     :
  </p>
  <div
    class="pl-4 mb-6"
    data-testid="moves-container"
  >
    <p>
      razor-wind
    </p>
  </div>
</div>
`;

exports[`On render, check that types is displayed correctly 1`] = `
<div>
  <p>
    <span
      class="font-bold underline text-lg"
    >
      Types
    </span>
     :
  </p>
  <div
    class="pl-4 mb-6"
    data-testid="types-container"
  >
    <p>
      grass
    </p>
    <p>
      poison
    </p>
  </div>
</div>
`;
